### PR TITLE
refactor(forms): deprecate unwanted control events aliases

### PR DESCRIPTION
**Pull request by JeanMeche**
The types changed in the commit shouldn't have been aliased in the first place.

### CODE REVIEW

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:631246215:BASE:631246242:1715044164405:d286eef9).

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
Justification for this change:

- this is an update to the feature that hasn't been released yet (will be a part of v18)
- during the RC phase, we've identified that there is an inconsistency in event class naming (introduced accidentally by aliasing correct symbols with incorrect names)
- this PR exposes correct symbols, while marking incorrect (existing) symbols as deprecated
- this allows to do a cleanup in Google's codebase in places where incorrect symbol names are used
- there is a followup PR that removes incorrect and deprecated symbols https://github.com/angular/angular/pull/55723

**Review by AndrewKushnir — APPROVED** [review]

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:631246215:BASE:631615839:1715132243074:87346d96).

**Review by thePunderWoman — APPROVED** [review]
reviewed-for: public-api

**Comment by AndrewKushnir** [comment]
@JeanMeche could you please rebase this PR and resolve merge conflicts when you get a chance?

File: goldens/public-api/forms/index.md [hunk 1]
```diff
@@ -800,14 +800,18 @@ export class PatternValidator extends AbstractValidatorDirective {
...
     constructor(pristine: boolean, source: AbstractControl);
     // (undocumented)
     readonly pristine: boolean;
     // (undocumented)
     readonly source: AbstractControl;
 }
 
+// @public @deprecated (undocumented)
+export class PristineEvent extends PristineChangeEvent {
+}
```
Comment (AndrewKushnir) [inline goldens/public-api/forms/index.md:813]: @JeanMeche could you please make `PristineEvent` a base class (and marked as `@deprecated`) and have `PristineChangeEvent` extend it? The problem is that we emit `PristineChangeEvent` event and the code that relies on `instanceof PristineEvent` check fails in some tests internally.

We'd need to have a model like this in the PR:
```
class A {}
class B extends A {}
const b = new B();  // emit this instance
console.log(b instanceof A);  // true
console.log(b instanceof B);  // true
```

Comment (JeanMeche) [reply goldens/public-api/forms/index.md]: Only for `PristineEvent` ? or should I do the same for `TouchedEvent`?

Comment (AndrewKushnir) [reply goldens/public-api/forms/index.md]: It'd be great to update all 3 events in the same way.

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:631246215:BASE:632379064:1715319326543:cb0148c8).

**Comment by AndrewKushnir** [comment]
@JeanMeche the presubmit is "green" 👍 The `test` CI job is failing due to a pre-existing payload size issue (that was fixed in the main branch) and symbol golden files that require an update. Could you please update golden files when you get a chance? Thank you.

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
**Caretaker note**: presubmit is "green", this PR is ready for merge.

**Comment by atscott** [comment]
This PR was merged into the repository by commit c8472e5e9e2cfc9cdfbdb0d8ca0254b2f7b4f287.

**Pull request closed without merging**